desc/compute/_stability.py

```diff
@@ -234,199 +236,191 @@ def _magnetic_well(params, transforms, profiles, data, **kwargs):
+    units="~",
+    units_long="None",
+    description="Parameter in ideal ballooning equation",
+    dim=2,
+    params=["Psi"],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["a", "g^aa", "g^ra", "g^rr", "|B|^2", "B^zeta", "iota", "shear", "rho"],
+    zeta0="array: points of vanishing integrated local shear to scan over. "
+    "Default 15 points linearly spaced in [-π/2,π/2]",
+)
+def _f_balloon(params, transforms, profiles, data, **kwargs):
+    zeta0 = kwargs.get("zeta0", jnp.linspace(-0.5 * jnp.pi, 0.5 * jnp.pi, 15))
+    zeta0 = zeta0.reshape(-1, 1)
+
+    psi_boundary = params["Psi"] / (2 * jnp.pi)
+    B_n = 2 * psi_boundary / data["a"] ** 2
+    constant1 = data["a"] * B_n**3
+    constant2 = data["a"] ** 3 * B_n
+
+    gds2 = (
+        data["rho"] ** 2 * data["g^aa"]
+        - 2
+        * data["rho"]
+        * jnp.sign(data["iota"])
+        * data["shear"]
+        * data["g^ra"]
+        * zeta0
+        + data["shear"] ** 2 * data["g^rr"] * zeta0**2
+    )
+    data["f ballooning"] = (constant1 / data["|B|^2"] / data["B^zeta"]) * gds2
+    data["g ballooning"] = (constant2 / data["|B|^2"] * data["B^zeta"]) * gds2
+    return data
+
+
+@register_compute_fun(
+    name="g ballooning",
+    # g = a³ Bₙ * b⋅∇ζ (dψ_N/dρ)² |∇α|² / B
+    label="a^3 B_n b \\cdot \\nabla ζ (\\partial_{\\rho} \\psi_N)^2 "
+    "\\vert \\nabla \\alpha \\vert^2 / B",
+    units="~",
+    units_long="None",
+    description="Parameter in ideal ballooning equation",
+    dim=2,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["f ballooning"],
+)
+def _g_balloon(params, transforms, profiles, data, **kwargs):
+    # noqa: unused dependency
+    return data
+
+
 @register_compute_fun(
     name="ideal ballooning lambda",
     label="\\lambda_{\\mathrm{ballooning}}=\\gamma^2",
     units="~",
     units_long="None",
-    description="Normalized squared ideal ballooning growth rate, "
-    "requires data along a field line",
-    dim=1,
-    params=["Psi"],
+    description="Normalized squared ideal ballooning growth rate",
+    dim=3,
+    params=[],
     transforms={"grid": []},
     profiles=[],
     coordinates="rtz",
-    data=[
-        "a",
-        "g^aa",
-        "g^ra",
-        "g^rr",
-        "cvdrift",
-        "cvdrift0",
-        "|B|",
-        "B^zeta",
-        "p_r",
-        "iota",
-        "shear",
-        "psi",
-        "psi_r",
-        "rho",
-    ],
+    data=["c ballooning", "f ballooning", "g ballooning"],
     source_grid_requirement={"coordinates": "raz", "is_meshgrid": True},
-    zeta0="array: points of vanishing integrated local shear to scan over. "
-    "Default 15 points linearly spaced in [-π/2,π/2]",
     Neigvals="int: number of largest eigenvalues to return, default value is 1.`"
     "If `Neigvals=2` eigenvalues are `[-1, 0, 1]` we get `[1, 0]`",
+    eigfuns="bool: Whether to return eigenfunctions. Default is true.",
```

Comment:
I think this probably isn't the best way to handle this, it should be its own compute function

Reply:
We would need to initiate a deprecation cycle. People are already using `ideal ballooning lambda` to compute eigenfunctions and I don't want to make rahul email them again for a change.

currently this pr does not interrupt their workflow.

Reply:
I like this option. Also, @f0uriest   the eigenfunction is still defined as a separate compute function so this is consistent with what you ask.

Reply:
doesn't this mean that if i do `eq.compute(["ballooning eigenvalues", "ballooning eigenfunctions"])` it will throw an error because I actually need to do `eq.compute(["ballooning eigenvalues", "ballooning eigenfunctions"], eigfuns=True)` which feels more confusing

Reply:
The default is `eigfuns=True`.

Reply:
It does not store the eigenvectors in memory when `eigfuns=False`